src/wallet/wallet.cpp

```diff
@@ -29,11 +29,42 @@
 #include <wallet/fees.h>
 
 #include <assert.h>
+#include <algorithm>
 #include <future>
 
 #include <boost/algorithm/string/replace.hpp>
 
-std::vector<CWalletRef> vpwallets;
+static std::vector<CWallet*> vpwallets;
+
+void AddWallet(CWallet* wallet)
```

Comment:
Do we want to provide guarantees that `GetWallets` returns no `nullptrs`? Could check here for that.

Reply:
Added assertions in `AddWallet` and `RemoveWallet`. Also added a check to prevent adding the same wallet (`RemoveWallet` already checks if wallet exists).